src/server/src/main/java/io/cassandrareaper/management/MetricsProxy.java

```diff
@@ -17,39 +17,149 @@
 import io.cassandrareaper.management.jmx.JmxMetricsProxy;
 
 import java.io.IOException;
 import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
 import javax.management.JMException;
 
+import com.google.common.collect.Lists;
+import org.joda.time.DateTime;
+
+
 public interface MetricsProxy {
 
-  static MetricsProxy create(ICassandraManagementProxy proxy) {
+  static MetricsProxy create(ICassandraManagementProxy proxy, Node node) {
     if (proxy == null) {
       throw new RuntimeException("ICassandraManagementProxy is null");
     }
     if (proxy instanceof JmxCassandraManagementProxy) {
-      return JmxMetricsProxy.create((JmxCassandraManagementProxy) proxy);
+      return JmxMetricsProxy.create((JmxCassandraManagementProxy) proxy, node);
     } else if (proxy instanceof HttpCassandraManagementProxy) {
-      return HttpMetricsProxy.create((HttpCassandraManagementProxy) proxy);
+      return HttpMetricsProxy.create((HttpCassandraManagementProxy) proxy, node);
     }
     throw new UnsupportedOperationException("Unknown ICassandraManagementProxy implementation");
   }
 
-  List<GenericMetric> collectTpStats(Node node) throws JMException, IOException;
+  List<GenericMetric> collectTpStats() throws JMException, IOException;
 
-  List<GenericMetric> collectDroppedMessages(Node node) throws JMException, IOException;
+  List<GenericMetric> collectDroppedMessages() throws JMException, IOException;
 
-  List<GenericMetric> collectLatencyMetrics(Node node) throws JMException, IOException;
+  List<GenericMetric> collectLatencyMetrics() throws JMException, IOException;
 
-  List<GenericMetric> collectGenericMetrics(Node node) throws JMException, IOException;
+  List<GenericMetric> collectGenericMetrics() throws JMException, IOException;
 
-  List<GenericMetric> collectPercentRepairedMetrics(Node node, String keyspaceName)
+  List<GenericMetric> collectPercentRepairedMetrics(String keyspaceName)
       throws JMException, IOException;
+
+  static List<GenericMetric> convertToGenericMetrics(Map<String, List<JmxStat>> jmxStats, Node node) {
+    List<GenericMetric> metrics = Lists.newArrayList();
+    DateTime now = DateTime.now();
+    for (Entry<String, List<JmxStat>> jmxStatEntry : jmxStats.entrySet()) {
+      for (JmxStat jmxStat : jmxStatEntry.getValue()) {
+        GenericMetric metric = GenericMetric.builder()
+            .withClusterName(node.getClusterName())
+            .withHost(node.getHostname())
+            .withMetricDomain(jmxStat.getDomain())
+            .withMetricType(jmxStat.getType())
+            .withMetricScope(jmxStat.getScope())
+            .withMetricName(jmxStat.getName())
+            .withMetricAttribute(jmxStat.getAttribute())
+            .withValue(jmxStat.getValue())
+            .withTs(now)
+            .build();
+        metrics.add(metric);
+      }
+    }
+
+    return metrics;
+  }
+
+  static ThreadPoolStat.Builder updateGenericMetricAttribute(GenericMetric stat,
```

Comment:
Issue: would be good to see a test for this. Copilot should be able to write something reasonable for it quite quickly I think.

Same for the other methods here.

Reply:
sounds good 👍

Reply:
Appears to have been added